.github/actions/branch-lock/src/main.ts

```diff
@@ -0,0 +1,104 @@
+import * as core from '@actions/core'
+import {actionInputs} from 'github-actions-utils'
+import {GraphQLClient, gql} from 'graphql-request'
+
+/**
+ * GitHub Actions entrypoint.
+ */
+export async function run(): Promise<void> {
+  try {
+    // Validate inputs
+    const lock = actionInputs.getBool('lock', true)
+    const branch = actionInputs.getString('branch', true)
+    const token = actionInputs.getString('token', true, true)
+    const owner = actionInputs.getString('owner', true)
+    const repo = actionInputs.getString('repo', true)
+
+    // Create github service
+    const githubService = new GitHubService(token)
+    const branchId = await githubService.searchBranchProtection(owner, repo, branch)
```

Comment:
This call already exists in oktokit https://github.com/octokit/plugin-rest-endpoint-methods.js/blob/main/docs/repos/getBranchProtection.md